pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/models/consumer/objects_api/membership/PNChannelMembership.java

```diff
@@ -20,31 +20,80 @@ public static class ChannelId {
 
     @Getter
     private final ChannelId channel;
+    @Getter
+    private final Map<String, Object> custom;
+    @Getter
+    private final String status;
+    @Getter
+    private final String type;
 
+    @Deprecated
     public static PNChannelMembership channel(final String channelId) {
         return new JustChannel(new ChannelId(channelId));
     }
 
+    @Deprecated
     public static PNChannelMembership channelWithCustom(final String channelId, final Map<String, Object> custom) {
-        return new ChannelWithCustom(new ChannelId(channelId), new HashMap<>(custom));
+        return new ChannelWithCustom(new ChannelId(channelId), new HashMap<>(custom), null, null);
+    }
+
+    // PNChannelMembership.PNChannelMembership(membership01ChannelId).type("t").status("s").custom(custom).build()
```

Comment:
instantiation using a function named the same as the type is a pattern in Kotlin which has top-level functions, but not in Java. In other places in java we have a pattern like this:
`SomeType.builder()` (static function)

Reply:
ok